src/wfs/src/main/java/org/vfny/geoserver/wfs/servlets/TestWfsPost.java

```diff
@@ -303,4 +316,24 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
             }
         }
     }
+
+    private void validateURL(HttpServletRequest request, String url) {
+        String proxyBase = GeoServerExtensions.getProperty(PROXY_BASE_URL);
+        if(proxyBase != null) {
+            if(!url.startsWith(proxyBase)) {
+                throw new IllegalArgumentException("Invalid url requested, the demo requests should be hitting: " + proxyBase);
+            }
+        }
+        
+        // use the requested url then, and remove the TestWfsPort
+        String requestedUrl = request.getRequestURL().toString();
+        // this should not happen, but let's not make it an open proxy if it does
+        if(!requestedUrl.endsWith(TEST_WFS_POST_PATH)) {
+            throw new IllegalStateException("Unepected, the TestWfsPost was accessed by a path not ending with TestWfsPost: " + requestedUrl);
+        }
+        String base = requestedUrl.substring(0, requestedUrl.lastIndexOf(TEST_WFS_POST_PATH));
+        if(!url.startsWith(base)) {
+            throw new IllegalArgumentException("Invalid url requested, the demo requests should be hitting: " + base);
+        }
```

Comment:
Uh you are probably right, best to put this part in the else of ``if(proxyBase != null)``
I'll make this quick change